pkg/reconciler/workload/resource/resource_controller.go

```diff
@@ -159,6 +160,15 @@ func filterNamespace(obj interface{}) bool {
 		logging.WithReconciler(klog.Background(), controllerName).WithValues("namespace", name).V(2).Info("skipping syncing Namespace")
 		return false
 	}
+	// Filter out namespaces which have been synced to a SyncTarget living on a KCP shard (typical e2e test situation)
```

Comment:
Even though this situation might happen most typically e2e tests, this is not limited to it.
It could happen in other cases as well.
Do we really want resources that are targets of a Syncing to become sources of another syncing ?
Do we want to enable this use-case ?
To me it seems better to disable this use-case until we clearly acknowledged its requirement, scoped its limits and explored its impacts.

Reply:
How could we opt-in the resource controller? Does this controller even do anything if there is no placement?

Reply:
Or is the problem that there is always a placement? Maybe we have to rethink our CWT structure, i.e. that the e2e does not create a fake pcluster that has workload APIs.

Reply:
@davidfestal I meant - if we have a (working) mechanism to opt in or out of syncing, let's use it?

Reply:
Yes, it does. It reconcile all the namespaces and resources, and ensures that the placement on individual resources match the placement on namespaces. That's really useless work.

Reply:
In addition to that, I've hit, while debugging, a case when such a resource has a `RequestState` label set to `Sync` while its  namespace doesn't. This leads to the resource reconciler patching the resource to suppress it.

Reply:
I split the PR to remove this questionable commit, and created a dedicated PR for it: https://github.com/kcp-dev/kcp/pull/2131 to continue the discussion